cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -229,6 +233,13 @@ public TwillController launch(TwillApplication twillApplication, Iterable<String
             // TokenSecureStoreUpdater.update() ignores parameters
             twillPreparer.addSecureStore(secureStoreUpdater.update(null, null));
           }
+
+          // Hack for CDAP-7021. Interacts with the patched YarnTwillPreparer class.
+          // we'll build jars for each restart of cdap master. Delete jars from any early cdap-master run.
```

Comment:
old comment before I realized it gets deleted when master starts up. Will remove.